### DISCUSSION TITLE
A little help required regarding FHIR DSTU2

### DISCUSSION THREAD

**Discussion by abdulazizibrahim** [nazrulworld/fhir.resources#87] [General]
Hi fellows,

I am new to the whole health-tech world and I had to work FHIR DSTU2 version. I was using this library to test out stuff and I faced some issues which I will addressing below and I hope you guys can help me out here.
 
`
from fhir.resources.DSTU2.subscription import Subscription

data = {

    "resourceType": "Subscription",

    "id": "1882f571-520c-4c07-ba41-9b83fdb4eb3d",

    "criteria": "ReferralRequest?_practitioner=Beck,Rachel MD&_encounter=Encounter/27300-AM0000120709",

   "reason": "callback",

    "channel": {

        "type": "rest-hook",

        "endpoint": "TBD",

        "payload": "application/json"

    }

}

sub = Subscription(**data)
`
 When I receive a subscription request (json), I want to create an FHIR Subscription object but when I try to do that I get the following error.
 `
ValidationError                           Traceback (most recent call last)
<ipython-input-12-1ec3841ca259> in <module>
----> 1 sub = Subscription(**data)

~/.local/lib/python3.9/site-packages/fhir/resources/DSTU2/fhirabstractmodel.py in __init__(__pydantic_self__, **data)
    134             )
    135         if errors:
--> 136             raise ValidationError(errors, __pydantic_self__.__class__)
    137 
    138         BaseModel.__init__(__pydantic_self__, **data)

ValidationError: 1 validation error for Subscription
resource_type
  Wrong ResourceType: ``fhir.resources.DSTU2.subscription.Subscription`` expects resource type ``Substance``, but got ``Subscription``. Make sure resource type name is correct and right ModelClass has been chosen. (type=value_error.wrong.resource_type; error=``fhir.resources.DSTU2.subscription.Subscription`` expects resource type ``Substance``, but got ``Subscription``. Make sure resource type name is correct and right ModelClass has been chosen.)
` 
I would really appreciate if you could help me out here. @nazrulworld

**Comment by nazrulworld** [comment]
@abdulazizibrahim I see it's not your fault in any way, it's a bug. Please open an issue, we will look into it.
 For now, you could make a simple override/patch to continue.

```
from fhir.resources.DSTU2.subscription import Subscription as OrgSubscription
class Subscription(OrgSubscription):
     resource_type = Field("Subscription", const=True)
```

Actuality DTU2 package is not properly tested. If possible You can check other classes as well.

**Reply by abdulazizibrahim** [reply]
@nazrulworld Yes, i will do that. Also I was looking through the repo and I think if you go to file `fhir.resources/fhir/resources/DSTU2/subscription.py` you will find ` resourse_type = Field("Substance", const=True )` on line 27. I think it may have something to do with it.
